### PR TITLE
Add Form

### PR DESCRIPTION
**Pull request by desorgui**
> ## Pull request changes
> -  Create a method called `current_user` in the ApplicationController to make current user data available to all controllers.
>     -  It returns the first user from the database.
> - Create forms to perform the following functions:
>     - Creates a Post on behalf of the `current_user` .
>     - Create a comment on behalf of the `current_user`.
> - Allow Users to add likes to Posts.

### CODE REVIEW

File: app/controllers/posts_controller.rb [hunk 1]
```diff
@@ -1,9 +1,22 @@
...
+    param_user_id = params['user_id'].to_i
+    @user_found = User.find_by(id: param_user_id)
+    @post = Post.find_by(id: param_id)
+    @all_comments = Comment.where(post_id: param_id)
+  end
+
+  def new
+    Post.create(title: params['title'], text: params['message'], author_id: params['author_id'])
+    redirect_to '/'
   end
```
Comment (mikethreels) [inline app/controllers/posts_controller.rb]: - [ ] the `new` method in the post_controller should just look like below
```
    def new
      @post = Post.new
    end

```

File: app/controllers/posts_controller.rb [hunk 2]
```diff
@@ -1,9 +1,22 @@
...
 
   def show
     puts params
+    param_id = params['id'].to_i
+    param_user_id = params['user_id'].to_i
+    @user_found = User.find_by(id: param_user_id)
+    @post = Post.find_by(id: param_id)
+    @all_comments = Comment.where(post_id: param_id)
+  end
```
Comment (mikethreels) [inline app/controllers/posts_controller.rb:16]: - [ ] there should be a create method in the postController

File: app/controllers/comments_controller.rb [hunk 3]
```diff
@@ -0,0 +1,11 @@
...
+  def index
+    @commented_post_id = params['id']
+  end
+
+  def comment
+    Comment.create(text: params['comment'], author_id: params['author_id'], post_id: params['post_id'])
+
+    redirect_to "/users/#{params['author_id']}/posts"
+  end
+end
```
Comment (mikethreels) [inline app/controllers/comments_controller.rb]: - [ ] you're supposed to use strong parameters for the params, have a look at [this lesson](https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/lessons/Controllers_lesson.md#:~:text=Pay%20special%20attention%20to%20the%20strong%20parameters%20description%20in%20Rails%20Guides.) for additional information.

File: app/controllers/likes_controller.rb [hunk 4]
```diff
@@ -0,0 +1,6 @@
+class LikesController < ApplicationController
+  def index
+    Like.create(author_id: params['author_id'], post_id: params['post_id'])
+    redirect_to '/'
+  end
```
Comment (mikethreels) [inline app/controllers/likes_controller.rb:5]: - [ ] you're supposed to use strong parameters for the params, have a look at [this lesson](https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/lessons/Controllers_lesson.md#:~:text=Pay%20special%20attention%20to%20the%20strong%20parameters%20description%20in%20Rails%20Guides.) for additional information.

**Review by mikethreels — CHANGES_REQUESTED** [review]
Hi Desor, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights
- linters are passing

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app/controllers/posts_controller.rb [hunk 5]
```diff
@@ -1,9 +1,26 @@
...
+
+  def new
+    Post.new
+  end
+
+  def create
+    Post.create(title: params['title'], text: params['message'], author_id: params['author_id'])
```
Comment (FrederickMih) [inline app/controllers/posts_controller.rb]: - [x] It's a requirement of this project to make use of strong parameters as requested by the previous reviewer. These parameters should be made private

```.ruby
private 
    def post_params
      params.require(:post).permit(:title, :text)
    end
```

File: app/controllers/posts_controller.rb [hunk 6]
```diff
@@ -1,9 +1,26 @@
...
+    @user_found = User.find_by(id: param_user_id)
+    @post = Post.find_by(id: param_id)
+    @all_comments = Comment.where(post_id: param_id)
+  end
+
+  def new
+    Post.new
+  end
+
+  def create
```
Comment (FrederickMih) [inline app/controllers/posts_controller.rb:22]: - [x] Based on the requirements of this project, a post is created on behalf of the currently logged-in user (current_user) of the app. You can simply implement it like this:



```ruby
  def create
      @post = @current_user.posts.new(post_params)

      respond_to do |format|
        if @post.save
          format.html { redirect_to @post }
        else
          format.html { render :new }
        end
      end
    end
```

File: app/controllers/comments_controller.rb [hunk 7]
```diff
@@ -0,0 +1,11 @@
+class CommentsController < ApplicationController
```
Comment (FrederickMih) [inline app/controllers/comments_controller.rb:1]: - [x] As requested by the previous reviewer, the comment controller, should equally have the new action. It should look like the piece of code below:

```ruby
  def new
    @comment = Comment.new
  end
```

File: app/controllers/comments_controller.rb [hunk 8]
```diff
@@ -0,0 +1,11 @@
...
+  def index
+    @commented_post_id = params['id']
+  end
+
+  def comment
+    Comment.create(text: params['comment'], author_id: params['author_id'], post_id: params['post_id'])
```
Comment (FrederickMih) [inline app/controllers/comments_controller.rb]: - [x] Based on the requirements of this project, a comment is created on behalf of the currently logged-in user. Hence let's fix this. Remember you implemented the current_user method in your application_controller.


- [x] As requested by the previous reviewer, you are supposed to make use of strong parameters. Kindly fix these issues.

**Review by FrederickMih — CHANGES_REQUESTED** [review]
Hi @desorgui , 

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.


 ## Highlights

- Correct Git flow :+1: 
- No Linter error :+1: 
- PR has a title and summary :+1: 

 ## Required Changes ♻️


_Check the comments under the review._

## Optional suggestions

- [Optional] Hi @desorgui, remember you had a #new action in your comment and post controllers. You'll equally require a view for creating both comments and posts. Hence you'll need a form in views/comments/new.html.erb and views/posts/new.html.erb. If you already added the forms let's kindly do renaming


_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by ShahierNashaat — APPROVED** [review]
Hi @desorgui, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## To Highlight 💯 

- Correct use of GitFlow ✔️
- Descriptive commit messages ✔️
- Professional looking pull request ✔️
- Your readme looks professional ✔️
- Your code is clean and well documented ✔️

# Approved :ok_hand: :ok_hand: :ok_hand:

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**